packages/aws-cdk/test/api/hotswap/appsync-mapping-templates-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,308 @@
+import { AppSync } from 'aws-sdk';
+import * as setup from './hotswap-test-setup';
+
+let hotswapMockSdkProvider: setup.HotswapMockSdkProvider;
+let mockUpdateResolver: (params: AppSync.UpdateResolverRequest) => AppSync.UpdateResolverResponse;
+let mockUpdateFunction: (params: AppSync.UpdateFunctionRequest) => AppSync.UpdateFunctionResponse;
+let mockListFunctions: (params: AppSync.ListFunctionsRequest) => AppSync.ListFunctionsResponse;
+
+beforeEach(() => {
+  hotswapMockSdkProvider = setup.setupHotswapTests();
+  mockUpdateResolver = jest.fn();
+  mockUpdateFunction = jest.fn();
+  mockListFunctions = jest.fn().mockReturnValue({ functions: [{ name: 'my-function', functionId: 'functionId' }] });
+  hotswapMockSdkProvider.stubAppSync(mockUpdateResolver, mockUpdateFunction, mockListFunctions);
+});
+
+test('returns undefined when a new Resolver is added to the Stack', async () => {
+  // GIVEN
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        AppSyncResolver: {
+          Type: 'AWS::AppSync::Resolver',
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+});
+
+test('calls the updateResolver() API when it receives only a mapping template difference in a Resolver', async () => {
+  // GIVEN
+  setup.setCurrentCfnStackTemplate({
+    Resources: {
+      AppSyncResolver: {
+        Type: 'AWS::AppSync::Resolver',
+        Properties: {
+          ApiId: 'apiId',
+          FieldName: 'myField',
+          TypeName: 'Query',
+          DataSourceName: 'my-datasource',
+          Kind: 'UNIT',
+          RequestMappingTemplate: '## original request template',
+          ResponseMappingTemplate: '## original response template',
+        },
+        Metadata: {
+          'aws:asset:path': 'old-path',
+        },
+      },
+    },
+  });
+  setup.pushStackResourceSummaries(setup.stackSummaryOf('AppSyncResolver', 'AWS::AppSync::Resolver', 'arn:aws:appsync:us-east-1:111111111111:apis/apiId/types/Query/resolvers/myField'));
```

Comment:
Maybe multi-line this puppy, for a little more readability?

```suggestion
  setup.pushStackResourceSummaries(setup.stackSummaryOf('AppSyncResolver', 'AWS::AppSync::Resolver',
    'arn:aws:appsync:us-east-1:111111111111:apis/apiId/types/Query/resolvers/myField'));
```